### PR TITLE
1kc-razer: rename to razer-unofficial

### PR DESCRIPTION
**Pull request by miccal**
https://github.com/orgs/Homebrew/discussions/4967

https://docs.brew.sh/Cask-Cookbook#special-affixes

### CODE REVIEW

**Comment by bevanjkay** [comment]
Is this an official application? It looks like the prefix is due to it being unofficial

File: Casks/r/razer.rb [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
   sha256 "9f57f928adb31ca843dbf2047a7d6ea308eceaccb50620792535d1fcd7ec73ca"
 
   url "https://github.com/1kc/razer-macos/releases/download/v#{version}/Razer.macOS-#{version}-universal.dmg"
   name "Razer macOS"
-  desc "Open source color effects manager for Razer devices"
-  homepage "https://github.com/1kc/razer-macos"
+  desc "Open source colour effects manager for Razer devices"
+  homepage "https://github.com/1kc/razer-macos/"
 
-  livecheck do
```
Comment (SMillerDev) [inline Casks/r/razer.rb:10]: Why not?

**Comment by SMillerDev** [comment]
> tl;dr I think it may have been renamed with the 'fork prefix' incorrectly; and it feels like either razer-unofficial; or better yet, the actual app name as marketed: razer-macos

`razer-unofficial` is fine if it doesn't conflict, but `razer-macos` is just as confusing while also going against our token guidelines.

**Comment by bevanjkay** [comment]
I would suggest that the correct solution here is not changing the token but that `brew search` returns correct results including the `name` stanza. 

According to `brew search --help` a casks `name` is only included in the search if you pass the `--desc` flag. So there is potentially room for improvement here. There is an existing issue in Homebrew/brew discussing handling searching of descriptions better, so noting this there could be helpful.

**Comment by SMillerDev** [comment]
Searching "razer" on https://formulae.brew.sh does surface this cask first.

**Comment by 0xdevalias** [comment]
> In this case I would suggest that 1kc-razer is the ideal token as it shows that it is an app called Razer developed by 1kc.

@bevanjkay It's technically an app called "Razer macOS", not "Razer"; developed by 1kc.

> I would suggest that the correct solution here is not changing the token but that brew search returns correct results including the name stanza.

@bevanjkay That would be ideal; as I'm sure this isn't the only cask that won't be easily searchable by its canonical name with the current limitations/bugs in the default search.

> According to brew search --help a casks name is only included in the search if you pass the --desc flag. So there is potentially room for improvement here.

@bevanjkay Agreed.

> There is an existing issue in Homebrew/brew discussing handling searching of descriptions better, so noting this there could be helpful.

@bevanjkay Are you able to link the relevant issue? If not I'll try and remember to find it when on a computer.

> Searching "razer" on https://formulae.brew.sh does surface this cask first.

@SMillerDev Fair; but that's not how I (and I suspect many others) search: from the terminal.

File: Casks/r/razer.rb (hunk 1 above)
Comment (miccal) [reply Casks/r/razer.rb]: Automatic via the `Git` strategy.

File: cask_renames.json [hunk 2]
```diff
@@ -11,5 +12,6 @@
   "opal": "opal-composer",
   "owasp-zap": "zap",
   "remotion": "multiapp",
-  "streamlabs-obs": "streamlabs"
+  "streamlabs-obs": "streamlabs",
+  "tea": "ossapp"
```
Comment (0xdevalias) [inline cask_renames.json:16]: Unrelated?

**Comment by bevanjkay** [comment]
The standard here is that we would use the vendor prefix, unless that doesn't achieve any less ambiguity.

So I still believe that token should remain `1kc-razer`.

https://docs.brew.sh/Acceptable-Casks#forks-and-apps-with-conflicting-names

**Comment by miccal** [comment]
I am done wasting my time dealing with this.

**Comment by bevanjkay** [comment]
@0xdevalias Thank you for your in-depth analysis of the situation here - https://github.com/orgs/Homebrew/discussions/4967

Regarding the issue in Homebrew/brew that could be somewhat related - https://github.com/Homebrew/brew/issues/16237
Parsing the `name` text (and `desc` text, as plain text) may help in this situation. But I would suggest that perhaps consideration could be given to including the `name` by default in `brew search` or adding a separate flag, as requiring the use of `--desc` to search the `name` is not immediately apparent.

**Comment by 0xdevalias** [comment]
> The standard here is that we would use the vendor prefix, unless that doesn't achieve any less ambiguity.
> 
> So I still believe that token should remain `1kc-razer`.
> 
> [docs.brew.sh/Acceptable-Casks#forks-and-apps-with-conflicting-names](https://docs.brew.sh/Acceptable-Casks#forks-and-apps-with-conflicting-names)

@bevanjkay It may just be that the way those docs are written is a little ambiguous, but it seems to be 2 different cases, merged into one (which is what I was trying to say earlier about it not being a fork ([Ref](https://github.com/Homebrew/homebrew-cask/pull/162185#issuecomment-1850039601))):

1. > Forks must have the vendor’s name as a prefix...
2. > For unrelated apps that share a name, the most popular one (usually the one already present) stays unprefixed. Since this can be subjective, if you disagree with a decision, open an issue and make your case to the maintainers.

As for my suggestion of keeping the `-macOS` suffix TODOREF, it was based on the 'exception to the rule' in this part:

- https://docs.brew.sh/Cask-Cookbook#finding-the-simplified-name-of-the-vendors-distribution
  - > Remove from the end: strings such as “Mac”, “for Mac”, “for OS X”, “macOS”, “for macOS”. These terms are generally added to ported software such as “MAME OS X.app”. Exception: when the software is not a port, and “Mac” is an inseparable part of the name, without which the name would be inherently nonsensical, as in [PlayOnMac.app](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/p/playonmac.rb).

And that the project specifically markets itself as 'Razer macOS', which matches user expectations in looking for it:

- https://github.com/1kc/razer-macos

image

image

The "macOS" suffix seems as much a part of it's name as anything else; it's not describing a port/etc; which is why I believe the exception (described above) is relevant.

That said; I would also be happy to accept `razer-unofficial`; and if the search stuff worked properly and matched on the `name`, then I probably would never have had a problem/thought to bring up renaming (or read so deeply into the specifics of the renaming rules) in the first place; so that would probably also make it a 'non-issue' for me.

---

> Thank you for your in-depth analysis of the situation here

@bevanjkay No problem :) I figured I would try and answer my own question first; or at least provide relevant context to help get a better answer.

> Regarding the issue in Homebrew/brew that could be somewhat related - [Homebrew/brew#16237](https://github.com/Homebrew/brew/issues/16237)
> Parsing the `name` text (and `desc` text, as plain text) may help in this situation. But I would suggest that perhaps consideration could be given to including the `name` by default in `brew search` or adding a separate flag, as requiring the use of `--desc` to search the `name` is not immediately apparent.

@bevanjkay _nods_ that makes sense. Thanks for the link :) I'll leave musings about the deeper specifics of that for once I've read the linked issue; and will probably add my thoughts there if I have any, as this PR doesn't seem like a good place for them not to get lost.

**Comment by bevanjkay** [comment]
> It may just be that the way those docs are written is a little ambiguous, but it seems to be 2 different cases, merged into one (which is what I was trying to say earlier about it not being a fork ([Ref](https://github.com/Homebrew/homebrew-cask/pull/162185#issuecomment-1850039601))):

I think the issue here is more to do with the way the docs link back to the information about "forks".

The cask cookbook is a bit more direct;

https://docs.brew.sh/Cask-Cookbook#potentially-misleading-name


> If the token for a piece of unofficial software that interacts with a popular service would make it look official and the vendor is not authorised to use the name, [a prefix must be added](https://docs.brew.sh/Acceptable-Casks#forks-and-apps-with-conflicting-names) for disambiguation.
In cases where the prefix is ambiguous and would make the app appear official, the -unofficial suffix may be used.

Therefore the vendor prefix is used first, then if it's still ambiguous the `-unofficial` suffix can be considered.

**Comment by 0xdevalias** [comment]
> Therefore the vendor prefix is used first

@bevanjkay If that is what the below docs are intending to suggest:

> [a prefix must be added](https://docs.brew.sh/Acceptable-Casks#forks-and-apps-with-conflicting-names) for disambiguation

Then they could definitely do with a refactor to make that clearer. As currently to me, that reads more like:

> If the token for a piece of unofficial software that interacts with a popular service would make it look official and the vendor is not authorised to use the name, [go read this](https://docs.brew.sh/Acceptable-Casks#forks-and-apps-with-conflicting-names) for rules about how to disambiguate it.

Phrased differently, as currently written, it comes across that the word "prefix" in the docs seems more of an arbitrary choice of word, and that the canonical rules are defined by the page it links to.
